app/lib/katello/resources/cdn.rb

```diff
@@ -1,7 +1,14 @@
 module Katello
   module Resources
     module CDN
-      SUPPORTED_SSL_VERSIONS = ['SSLv23', 'TLSv1'].freeze
+      SUPPORTED_TLS_VERSIONS_MAPPING = {
+        'TLSv1' => OpenSSL::SSL::TLS1_VERSION,
```

Comment:
I wonder if we need the same if defined check for this older version in the event that a Ruby version removes it.

Reply:
I suppose there is no reason I couldn't include the relevant check for each protocol version, just to future proof everything.

Reply:
I pushed an update that demonstrates what that looks like.